test/repo_test.rb

```diff
@@ -524,11 +518,13 @@ def test_init_with_wrong_argument
 class RepositoryCloneTest < Rugged::TestCase
   def setup
     @tmppath = Dir.mktmpdir
-    @source_path = "file://" + File.join(Rugged::TestCase::TEST_DIR, 'fixtures', 'testrepo.git')
-  end
+    Rugged::TestCase::FixtureRepo.ensure_cleanup @tmppath
```

Comment:
Making all temporary directory removal go through the same procedure for working around the open file handle issue mentioned later.